argoverse/data_loading/argoverse_tracking_loader.py

```diff
@@ -171,16 +171,15 @@ def image_list_sync(self) -> Dict[str, List[np.ndarray]]:
                     self._image_timestamp_list_sync[log][camera] = cast(
                         List[int],
                         list(
-                            filter(
-                                lambda x: x is not None,
-                                (
-                                    self.sync.get_closest_cam_channel_timestamp(x, camera, log)
-                                    for x in self._lidar_timestamp_list[log]
-                                ),
-                            )
+                            self.sync.get_closest_cam_channel_timestamp(x, camera, log)
+                            for x in self._lidar_timestamp_list[log]
                         ),
                     )
 
+                    assert len(self._image_timestamp_list_sync[log][camera]) == len(
```

Comment:
Thanks for this PR, Patsorn. Can we check on which logs from train1/train2/train3/val/test this assert passes? Another option would be to make an argument to the constructor that states in this rare edge case I prefer "None" or the closest image forwards or backwards, or the closest stale image (backwards only)

Reply:
That assertion always pass, even when we have a missing image (we replace the missing image with None or closest image), it's just a sanity check :) 
I added the option to choose what to return when there are missing image, current version of argoverse dataset doesn't have anything missing in the ring camera, though there are some stereo image timestamp missing, which hopefully this option would be useful.